scripts/pylib/twister/twisterlib/harness.py

```diff
@@ -162,6 +163,14 @@ class Console(Harness):
 
     def configure(self, instance):
         super(Console, self).configure(instance)
+        if self.regex is None or len(self.regex) == 0:
+            self.state = "failed"
```

Comment:
if the definition of the testcase is wrong, this should be treated as an error. So, instead of doing this here and calling this a fail, error already during the parsing of the test (in config_parser.py)

Reply:
> i wonder if this can be dealt with on the schema level, i.e.. disallowing empty sequences.

pykwalify has `allowempty` but for `map' type only
https://pykwalify.readthedocs.io/en/unstable/validation-rules.html#allowempty

Reply:
> if the definition of the testcase is wrong, this should be treated as an error.

changed to 'error' status for instances

>  So, instead of doing this here and calling this a fail, error already during the parsing of the test (in config_parser.py)

it seems not easy to avoid these checks here with the current twister harness design: `testsute-schema.yaml` defines common `harness_config` for all types of harnesses simultaneously and parser checks for its presence with allowed (not required) properties. Later harness `configure()` needs to check what it got from the test instance.